packages/site/src/layout/ComponentView.tsx

```diff
@@ -46,147 +103,139 @@ export const ComponentView = () => {
     };
   }, []);
 
-  const ComponentContent = PageMeta?.content;
+  const ComponentContent = getComponentContent(PageMeta, type);
 
-  const code =
-    type === "web" && PageMeta?.component?.element
-      ? PageMeta?.component?.element
-      : PageMeta?.component?.mobileElement;
+  const code = getComponentElement(PageMeta, type);
 
   useEffect(() => {
-    if (iframe?.current || iframeMobile?.current) {
+    if (code) {
       setTimeout(() => updateCode(code as string, true), 100);
     }
-  }, [code, type]);
+  }, [code, type, updateCode]);
```

Comment:
this was the cause of the issue where updating the code in the little sandbox area wouldn't update the example.

updateCode wasn't stable, so it created a loop preventing the code from properly updating.

technically we could omit `updateCode` but that's not great. we should have the correct deps for the `useEffect`. so instead, I've stabilized `updateCode` since it's a `useCallback` anyway.

I'm pretty satisfied with this set up. the only other alternative I can think of at the moment would be to avoid the `useEffect` entirely. make it more explicit and event driven, if that's even an option.